src/test/scala/org/constellation/domain/transaction/TransactionValidatorTest.scala

```diff
@@ -33,6 +36,65 @@ class TransactionValidatorTest
     transactionValidator = new TransactionValidator[IO](transactionService)
   }
 
+  "validateDestinationAddress" - {
+    val validAddress = Fixtures.id.address
+
+    "should fail validation" - {
+      "when destination address is smaller than 30 characters" in {
+        val tx = createTransaction("sender", "aabbcc", LastTransactionRef.empty, LastTransactionRef.empty)
+        val result = TransactionValidator.validateDestinationAddress(tx)
+
+        result shouldBe Validated.invalidNel(InvalidDestinationAddress(tx))
+      }
+
+      "when destination address contains non base58 character" in {
+        val tx = createTransaction("sender", validAddress + "%", LastTransactionRef.empty, LastTransactionRef.empty)
```

Comment:
Won't it fail currently because of the length condition and not base58 characters condition? It's 40 chars long address + 1 char